packages/opencensus-web-instrumentation-zone/src/interaction-tracker.ts

```diff
@@ -170,18 +176,23 @@ export class InteractionTracker {
 
     // Timeout to reset currentEventTracingZone to allow the creation of a new
     // zone for a new user interaction.
-    Zone.root.run(() =>
-      setTimeout(
-        () => (this.currentEventTracingZone = undefined),
+    Zone.root.run(() => {
+      // Store the timeout in case the interaction finishes before
+      // this callback runs.
+      this.currentResetTracingZoneTimeout = window.setTimeout(
+        () => this.resetCurrentTracingZone(),
         RESET_TRACING_ZONE_DELAY
-      )
-    );
-    console.log('New zone:');
-    console.log(this.currentEventTracingZone);
+      );
+    });
+  }
+
+  resetCurrentTracingZone() {
+    this.currentEventTracingZone = undefined;
+    this.currentResetTracingZoneTimeout = undefined;
   }
 
   /** Increments the count of outstanding tasks for a given interaction id. */
-  private incrementTaskCount(interactionId: string) {
+  incrementTaskCount(interactionId: string) {
```

Comment:
The motivation is to make sure we actually count the correct amount of tasks involved in the interaction. The duration of interaction is a good test but might give a wrong testing on the amount of tracked tasks for every interaction, as we actually want to check that, for example, macrotasks are handled. I am open to different approach though.

Reply:
Well, I guess to me the fact that it is counting tasks is more of an implementation detail that is not exposed to the end user of the tracker. So for an integration level test, I think we should be testing the surface as experienced by the user as close as possible.

Another idea would be to try to split out the part of the code that does the task counting and test that separately more as an implementation detail to make sure the counts are correct. And then in the integration test to just measure the total time and span name/metadata.

Reply:
Final decision: I decided to stop testing the task count as it should be more an implementation testing more than integration testing. Also, this could be done in a future refactoring on `InteractinTracker` to test every specific module.